python/libkvikio/cmake/thirdparty/get_nvcomp.cmake

```diff
@@ -0,0 +1,35 @@
+# =============================================================================
+# Copyright (c) 2021-2025, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+# =============================================================================
+
+set(KVIKIO_USE_PROPRIETARY_BINARY ON)
+
+# This function finds nvcomp and sets any additional necessary environment variables.
+function(find_and_configure_nvcomp)
+
+  include(${rapids-cmake-dir}/cpm/nvcomp.cmake)
+  set(export_args)
+  if(KvikIO_EXPORT_NVCOMP)
+    # We're vendoring nvcomp and we only want `libnvcomp.so.4`
+    set(export_args BUILD_EXPORT_SET nvcomp)
+  endif()
+
+  rapids_cpm_nvcomp(${export_args} USE_PROPRIETARY_BINARY ${KVIKIO_USE_PROPRIETARY_BINARY})
+
+  # Per-thread default stream
+  if(TARGET nvcomp AND PER_THREAD_DEFAULT_STREAM)
```

Comment:
I don't see `PER_THREAD_DEFAULT_STREAM` used or documented anywhere else. It's not strictly relevant to this PR, but should we go ahead and add an explicit top-level option for this?